sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/simple/CastTests.java

```diff
@@ -189,5 +189,124 @@ public void testFailingTimeCasts() {
                 "Cast function cannot convert value of type INTEGER to type DATE");
         this.statementsFailingInCompilation("CREATE VIEW V AS SELECT CAST(DATE '2024-01-01' AS INTEGER)",
                 "Cast function cannot convert value of type DATE to type INTEGER");
+
+        this.statementsFailingInCompilation("CREATE VIEW V AS SELECT CAST(X'01' AS TIME)",
+                "Conversion of BINARY object to ");
+    }
+
+    @Test
+    public void testAllCasts() {
```

Comment:
Does this also test casts for nullable types?

Reply:
Quote from the paragraph above:
> (We exercise only 1 of the 4 variants for each cast for now, without varying nullability)

Reply:
it's not easy to write a cast in SQL that throws away nullability - you really have to rely on the calcite type inference to infer that a type is not nullable.